packages/next/src/server/next.ts

```diff
@@ -252,7 +252,7 @@ function createServer(options: NextServerOptions): NextServer {
 
 // Support commonjs `require('next')`
 module.exports = createServer
-exports = module.exports
+// exports = module.exports
```

Comment:
Seems this is a bug?

Reply:
The problem is that when it's executed, `exports` is overridden.

```js
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    NextServer: function() {
        return NextServer;
    },
    // Support `import next from 'next'`
    default: function() {
        return _default;
    }
});
```

This is the generated code for exports, and those are removed by

```js
// Support commonjs `require('next')`
module.exports = createServer;
exports = module.exports;
const _default = createServer;

```

at the end of the file.

Reply:
But it's not a bug. ESM mixed with common js does not have the _correct_ behavior.